modules/core/src/lib/attribute.js

```diff
@@ -287,36 +288,40 @@ export default class Attribute extends BaseAttribute {
 
   // Use external buffer
   // Returns true if successful
-  setExternalBuffer(buffer, numInstances) {
+  setExternalBuffer(buffer) {
     const state = this.userData;
 
     if (buffer) {
       state.isExternalBuffer = true;
       this.clearNeedsUpdate();
 
+      if (ArrayBuffer.isView(buffer)) {
+        buffer = {value: buffer};
+      }
+
       if (buffer instanceof Buffer) {
         if (this.externalBuffer !== buffer) {
           this.update({constant: false, buffer});
           state.needsRedraw = true;
         }
-      } else if (this.value !== buffer) {
-        if (!ArrayBuffer.isView(buffer)) {
+      } else if (this.value !== buffer.value) {
+        const attribute = buffer;
+        if (!ArrayBuffer.isView(attribute.value)) {
           throw new Error('Attribute prop must be typed array');
         }
-        if (state.auto && buffer.length <= numInstances * this.size) {
-          throw new Error('Attribute prop array must match length and size');
-        }
 
         const ArrayType = glArrayFromType(this.type || GL.FLOAT);
-        if (buffer instanceof ArrayType) {
-          this.update({constant: false, value: buffer});
+        if (attribute.value instanceof ArrayType) {
```

Comment:
The catch here is that if an external attribute overwrote the default `size` `offset` etc. it will not be automatically reverted when the external buffer is unset. Arguably this is an advanced feature and the user should not reuse the same layer with both automatically and externally generated buffer...